packages/cdk-assets/lib/private/handlers/files.ts

```diff
@@ -57,7 +58,7 @@ export class FileAssetHandler implements IAssetHandler {
 
   private async packageFile(): Promise<string> {
     const source = this.asset.source;
-    const fullPath = path.resolve(this.workDir, this.asset.source.path);
+    const fullPath = isExternalSource(source) ? await shell(source.executable.split(' ')) : path.resolve(this.workDir, source.path);
```

Comment:
Again I'd rather we take two obviously different paths based on the asset type.

Recall the image I sent you:

![image](https://user-images.githubusercontent.com/524162/103521775-99023480-4e79-11eb-9598-dd07b0775eb5.png)

Right now this is implementing the 2nd pipeline, but I think the 1st pipeline actually makes a lot more sense.